### PR TITLE
Go 1.18 support

### PR DESCRIPTION
**Pull request by aarzilli**

### CODE REVIEW

**Comment by aarzilli** [comment]
cc @heschi I think you wanted to see this
1. tests on arm64 are disabled, tracking the regabi conversion as it's happening seems like too much of a hassle
2. generic support is included, except for setting a breakpoint on a generic function **by function name**, because of https://github.com/golang/go/issues/48032

**Comment by heschi** [comment]
Thank you very much. We're working on an internal generics preview program and having Delve work would be of great help.

File: pkg/proc/bininfo.go [hunk 1]
```diff
@@ -149,21 +149,156 @@ func (err *ErrFunctionNotFound) Error() string {
...
+		for _, pc := range bi.inlinedCallLines[fileLine{filename, lineno}] {
+			pcs = append(pcs, line.PCStmt{PC: pc, Stmt: true})
+		}
 	}
+
+	if len(pcs) == 0 {
+		return nil, &ErrCouldNotFindLine{fileFound, filename, lineno}
+	}
+
+	// 2. assign all occurences of filename:lineno to their containing function
```
Comment (derekparker) [inline pkg/proc/bininfo.go:186]: I don't see a `1.`, but there's a `2.` and `.3`, etc... in this comment list.

File: _scripts/test_linux.sh [hunk 2]
```diff
@@ -21,7 +21,10 @@ function getgo {
 }
 
 if [ "$version" = "gotip" ]; then
-	exit 0
+	# TODO: remove this
```
Comment (derekparker) [inline _scripts/test_linux.sh:24]: This will be removed once the regabi stuff is a little less chaotic I assume?

**Review by derekparker — APPROVED** [review]
LGTM—just a few nits and questions.

File: pkg/proc/bininfo.go (hunk 1 above)
Comment (aarzilli) [reply pkg/proc/bininfo.go:188]: Added.

File: _scripts/test_linux.sh (hunk 2 above)
Comment (aarzilli) [reply _scripts/test_linux.sh:24]: Yes, I'm thinking of doing it on the timeline we'd normally support the next version of Go, remove this and fix the test failures.

**Pull request merged**